File: backend/backend/urls.py
from main.views import (
    UserCreateView, UserDetailView, UserInfoView, UserSearchView, UpdateOnlineStatusView,
    GroupListCreateView, GroupDetailView, CreatePrivateChatView, GroupMembersView, LeaveGroupView,
    MessageListCreateView, MessageDetailView, MessageStatusView, MarkMessagesReadView,
    BlockedUsersListView, BlockUserView, UnblockUserView, AIConfigurationView, AIConfigTestView
)
from rest_framework_simplejwt.views import (
    TokenObtainPairView,
    TokenBlacklistView,
    TokenRefreshView,
)
from django.urls import path, include
from django.contrib import admin
from django.conf import settings 
from django.conf.urls.static import static 

urlpatterns = [
    path("admin/", admin.site.urls),
    
    # Authentication
    path("api/token/", TokenObtainPairView.as_view(), name="token_obtain_pair"),
    path("api/token/refresh/", TokenRefreshView.as_view(), name="token_refresh"),
    path("api/token/blacklist/", TokenBlacklistView.as_view(), name="token_blacklist"),
    
    # User management
    path("api/users/register/", UserCreateView.as_view(), name="user_register"),
    path("api/users/profile/", UserDetailView.as_view(), name="user_profile"),
    path("api/users/info/", UserInfoView.as_view(), name="user_info"),
    path("api/users/search/", UserSearchView.as_view(), name="user_search"),
    path("api/users/status/", UpdateOnlineStatusView.as_view(), name="update_status"),
    
    # Groups and chats
    path("api/groups/", GroupListCreateView.as_view(), name="group_list_create"),
    path("api/groups/<int:pk>/", GroupDetailView.as_view(), name="group_detail"),
    path("api/groups/private/create/", CreatePrivateChatView.as_view(), name="create_private_chat"),
    path("api/groups/<int:group_id>/members/", GroupMembersView.as_view(), name="group_members"),
    path("api/groups/<int:group_id>/leave/", LeaveGroupView.as_view(), name="leave_group"),
      # Messages
    path("api/groups/<int:group_id>/messages/", MessageListCreateView.as_view(), name="message_list_create"),
    path("api/groups/<int:group_id>/messages/read/", MarkMessagesReadView.as_view(), name="mark_messages_read"),
    path("api/messages/<int:pk>/", MessageDetailView.as_view(), name="message_detail"),
    path("api/messages/<int:message_id>/status/", MessageStatusView.as_view(), name="message_status"),    # Blocking
    path("api/blocked-users/", BlockedUsersListView.as_view(), name="blocked_users"),
    path("api/users/blocked/", BlockedUsersListView.as_view(), name="blocked_users_alt"),
    path("api/users/block/", BlockUserView.as_view(), name="block_user"),
    path("api/users/unblock/", UnblockUserView.as_view(), name="unblock_user"),
    path("api/users/unblock/<int:user_id>/", UnblockUserView.as_view(), name="unblock_user_by_id"),
      # AI Configuration
    path("api/ai/config/", AIConfigurationView.as_view(), name="ai_config"),
    path("api/ai/config/test/", AIConfigTestView.as_view(), name="ai_config_test"),
]

if settings.DEBUG:
    urlpatterns += static(settings.MEDIA_URL, document_root=settings.MEDIA_ROOT)

File: frontend/src/api/urls.js
const API_BASE_URL = 'http://192.168.1.101:8000/api';
const WS_BASE_URL = 'ws://192.168.1.101:8000';

export const API_URLS = {
    BASE: API_BASE_URL,
    
    // Auth
    LOGIN: `${API_BASE_URL}/token/`,
    REFRESH_TOKEN: `${API_BASE_URL}/token/refresh/`,
    REGISTER: `${API_BASE_URL}/users/register/`,
    LOGOUT: `${API_BASE_URL}/token/blacklist/`,
    
    // User
    USER_INFO: `${API_BASE_URL}/users/info/`,
    USER_PROFILE: `${API_BASE_URL}/users/profile/`,
    USER_SEARCH: (query) => `${API_BASE_URL}/users/search/?q=${query}`,
    BLOCK_USER: `${API_BASE_URL}/users/block/`,
    UNBLOCK_USER: (userId) => `${API_BASE_URL}/users/unblock/${userId}/`,
    BLOCKED_USERS: `${API_BASE_URL}/users/blocked/`,
      // Groups & Chats
    GROUPS: `${API_BASE_URL}/groups/`,
    CREATE_PRIVATE_CHAT: `${API_BASE_URL}/groups/private/create/`,
    GROUP_DETAIL: (groupId) => `${API_BASE_URL}/groups/${groupId}/`,
    LEAVE_GROUP: (groupId) => `${API_BASE_URL}/groups/${groupId}/leave/`,
    DELETE_GROUP: (groupId) => `${API_BASE_URL}/groups/${groupId}/`,
    REMOVE_MEMBER: (groupId) => `${API_BASE_URL}/groups/${groupId}/remove-member/`,
      // Messages
    MESSAGES: (groupId) => `${API_BASE_URL}/groups/${groupId}/messages/`,
    DELETE_MESSAGE: (messageId) => `${API_BASE_URL}/messages/${messageId}/`,
    MESSAGE_STATUS: (messageId) => `${API_BASE_URL}/messages/${messageId}/status/`,
    MARK_MESSAGES_READ: (groupId) => `${API_BASE_URL}/groups/${groupId}/messages/read/`,
    
    // AI
    AI_CONFIG: `${API_BASE_URL}/ai/config/`,
};

export const WEBSOCKET_URLS = {
    NOTIFICATIONS: `${WS_BASE_URL}/ws/notifications/`,
    CHAT: (groupId) => `${WS_BASE_URL}/ws/chat/${groupId}/`,
};